File: core/src/main/java/com/starshootercity/abilities/Vegetarian.java
package com.starshootercity.abilities;

import com.starshootercity.OriginSwapper;
import net.kyori.adventure.key.Key;
import org.bukkit.Material;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.event.player.PlayerItemConsumeEvent;
import org.bukkit.potion.PotionEffect;
import org.bukkit.potion.PotionEffectType;
import org.jetbrains.annotations.NotNull;

import java.util.ArrayList;
import java.util.List;

public class Vegetarian implements VisibleAbility, Listener {
    List<Material> meat = new ArrayList<>() {{
        add(Material.PORKCHOP);
        add(Material.COOKED_PORKCHOP);
        add(Material.BEEF);
        add(Material.COOKED_BEEF);
        add(Material.CHICKEN);
        add(Material.COOKED_CHICKEN);
        add(Material.RABBIT);
        add(Material.COOKED_RABBIT);
        add(Material.MUTTON);
        add(Material.COOKED_MUTTON);
        add(Material.RABBIT_STEW);
        add(Material.COD);
        add(Material.COOKED_COD);
        add(Material.TROPICAL_FISH);
        add(Material.SALMON);
        add(Material.COOKED_SALMON);
        add(Material.PUFFERFISH);
    }};

    @EventHandler
    public void onPlayerItemConsume(PlayerItemConsumeEvent event) {
        if (event.getItem().getType() == Material.POTION) return;
        AbilityRegister.runForAbility(event.getPlayer(), getKey(), () -> {
            if (meat.contains(event.getItem().getType())) {
                event.setCancelled(true);
                event.getItem().setAmount(event.getItem().getAmount() - 1);
                event.getPlayer().addPotionEffect(new PotionEffect(PotionEffectType.POISON, 300, 1, false, true));
            }
        });
    }
    @Override
    public @NotNull List<OriginSwapper.LineData.LineComponent> getDescription() {
        return OriginSwapper.LineData.makeLineFor(
                "You can't digest any meat.",
                OriginSwapper.LineData.LineComponent.LineType.DESCRIPTION);
    }

    @Override
    public @NotNull List<OriginSwapper.LineData.LineComponent> getTitle() {
        return OriginSwapper.LineData.makeLineFor(
                "Vegetarian",
                OriginSwapper.LineData.LineComponent.LineType.TITLE
        );
    }

    @Override
    public @NotNull Key getKey() {
        return Key.key("origins:vegetarian");
    }
}

File: core/src/main/kotlin/ru/turbovadim/abilities/Vegetarian.kt
package ru.turbovadim.abilities

import ru.turbovadim.OriginSwapper.LineData.Companion.makeLineFor
import ru.turbovadim.OriginSwapper.LineData.LineComponent
import net.kyori.adventure.key.Key
import org.bukkit.Material
import org.bukkit.event.EventHandler
import org.bukkit.event.Listener
import org.bukkit.event.player.PlayerItemConsumeEvent
import org.bukkit.potion.PotionEffect
import org.bukkit.potion.PotionEffectType

class Vegetarian : VisibleAbility, Listener {

    var meat: MutableList<Material> = mutableListOf(
        Material.PORKCHOP,
        Material.COOKED_PORKCHOP,
        Material.BEEF,
        Material.COOKED_BEEF,
        Material.CHICKEN,
        Material.COOKED_CHICKEN,
        Material.RABBIT,
        Material.COOKED_RABBIT,
        Material.MUTTON,
        Material.COOKED_MUTTON,
        Material.RABBIT_STEW,
        Material.COD,
        Material.COOKED_COD,
        Material.TROPICAL_FISH,
        Material.SALMON,
        Material.COOKED_SALMON,
        Material.PUFFERFISH
    )

    @EventHandler
    fun onPlayerItemConsume(event: PlayerItemConsumeEvent) {
        if (event.item.type == Material.POTION) return

        val player = event.player
        runForAbility(player) { p ->
            if (meat.contains(event.item.type)) {
                event.isCancelled = true
                event.item.amount -= 1
                p.addPotionEffect(PotionEffect(PotionEffectType.POISON, 300, 1, false, true))
            }
        }
    }

    override val description: MutableList<LineComponent> = makeLineFor(
        "You can't digest any meat.",
        LineComponent.LineType.DESCRIPTION
    )

    override val title: MutableList<LineComponent> = makeLineFor(
        "Vegetarian",
        LineComponent.LineType.TITLE
    )

    override fun getKey(): Key {
        return Key.key("origins:vegetarian")
    }
}
